api/src/main/java/org/apache/iceberg/NaNOnlyFieldMetrics.java

```diff
@@ -17,51 +17,50 @@
  * under the License.
  */
 
-package org.apache.iceberg.parquet;
+package org.apache.iceberg;
 
 import java.nio.ByteBuffer;
-import org.apache.iceberg.FieldMetrics;
 
 /**
- * Iceberg internally tracked field level metrics, used by Parquet writer only.
+ * Iceberg internally tracked field level metrics, used by Parquet and ORC writers only.
  * <p>
- * Parquet keeps track of most metrics in its footer, and only NaN counter is actually tracked by writers.
- * This wrapper ensures that metrics not being updated by Parquet writers will not be incorrectly used, by throwing
+ * Parquet/ORC keeps track of most metrics in file statistics, and only NaN counter is actually tracked by writers.
+ * This wrapper ensures that metrics not being updated by those writers will not be incorrectly used, by throwing
  * exceptions when they are accessed.
  */
-public class ParquetFieldMetrics extends FieldMetrics {
```

Comment:
I thought the reason for having `ParquetFieldMetrics` extending `FieldMetrics` is to allow ORC and parquet to diverge if needed. If we are moving them back to a common class, why not just move everything back to `FieldMetrics`?

Reply:
It's actually to allow Parquet and ORC to diverge from Avro, so we will use the actual `FiledMetrics` in avro

Reply:
I am a bit confused, in that case why not have 3 separated classes `ParquetFieldMetrics`, `ORCFieldMetrics` and `AvroFieldMetrics`?

Reply:
Yeah that's also an alternative approach, I wasn't sure if I want to directly duplicate the code in `ParquetFieldMetrics` to create a ORC version though since I don't think parquet/ORC libraries will support NaN natively soon.

Reply:
I don't see a need for 3 classes when 2 of them would be nearly identical. I like what is in this PR, with the one note about naming.